### PR TITLE
feat: modularization

### PR DESCRIPTION
**Pull request by caipre**
This is a rather large commit, but most of the changes are structural.
We introduce a few new targets (and directories in the repository) to
support modularization of the app. In this commit, we're extracting the
`HomeFeature` from the `App` target, encapsulating its state, actions,
and view into a separate target. This means we need a way to transform
the `Store<AppState, AppAction>` to a `Store<HomeState, HomeAction>`,
which is accomplished by the new `Store#view` method.

Most of the other features still need to be extracted, but the process
will be easier now that it's been done once. The steps amount to:

  1. Create targets and associated directories for the feature
  2. Create models for the state and actions of the feature
  3. Remove any dependencies from the feature to the app
  4. Wire the feature into the app using the `Store#view` projection

We'll need some inversion of control code to make this work for the
navigation links in `HomeFeature`, since ideally that feature doesn't
know about the others, even after they've been modularized.

### CODE REVIEW

File: MotDuJour/Features/Home/Sources/HomeView.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            store.send(.didTapLanguage(language: selection))
+          }
+        TextField("Search", text: $searchQuery)
+          .background(Color.gray)
+          .onChange(of: searchQuery) { query in
+            store.send(.didTapSearch(query: query))
+          }
+//        NavigationLink("today's word".localizedCapitalized, destination: WordDefinitionView())
+//        NavigationLink("Recently viewed".localizedCapitalized, destination: RecentsView())
+//        NavigationLink("Favorites".localizedCapitalized, destination: FavoritesView(store: store))
```
Comment (caipre) [inline MotDuJour/Features/Home/Sources/HomeView.swift:34]: Todo: Inversion of Control

File: MotDuJour/Project.swift [hunk 2]
```diff
@@ -21,7 +21,7 @@ let infop: InfoPlist = .extendingDefault(with: [
 ])
 
 let p = Project(
-  name: "ComposableArchitecture",
+  name: "MotDuJour",
```
Comment (caipre) [inline MotDuJour/Project.swift:24]: The application is now named "Mot du Jour", and we have an internal library named "ComposableArchitecture."

File: MotDuJour/ComposableArchitecture/Sources/Store.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  ) -> Store<LocalValue, LocalAction> {
+    let localStore = Store<LocalValue, LocalAction>(
+      with: toLocalValue(self.value),
+      reducer: { (localValue, localAction) in
+        self.send(toGlobalAction(localAction))
+        localValue = toLocalValue(self.value)
+      }
+    )
+    return localStore
+  }
```
Comment (caipre) [inline MotDuJour/ComposableArchitecture/Sources/Store.swift:32]: A way to transform a `Store<Value, Action>` to a narrower view of the store, for use by features.

**Comment by joelevin** [comment]
+1

**Pull request merged**